### ISSUE TITLE
Regression- find_in_parent_folders with fallback not looking into parent folders

### ISSUE THREAD

**Issue by kruttik-lab49** [gruntwork-io/terragrunt#1210]
Hi,

I am migrating my terragrunt source from `v0.18.7` to `0.23.23`. In the previous version I was able to use something like this-

```hcl
"${get_tfvars_dir()}/${find_in_parent_folders("env.tfvars", "DO_NOT_USE")}"
```
In this case I had a single `DO_NOT_USE.tfvars` at the root of the project and if the `env.tfvars` file was not found, it correctly returned the fallback file.

I changed this to as per the [reference repo](https://github.com/gruntwork-io/terragrunt-infrastructure-live-example/blob/master/terragrunt.hcl)-
```hcl
read_terragrunt_config(find_in_parent_folders("env.hcl", "DO_NOT_USE.hcl"))
```
but now I get an error 
```sh
Call to function "read_terragrunt_config" failed: Terragrunt config <terragrunt-repo>/<modulepath>/<some-module>/DO_NOT_USE.hcl not found.
```
The `DO_NOT_USE.hcl` file is in the `<terragrunt-repo>` and I was expecting the fallback file to also be reached in the parent folders and not just in the module/include folder.

Is this a regression or am I missing something?

__Update__
Looking at the example in [docs](https://terragrunt.gruntwork.io/docs/reference/built-in-functions/#read_terragrunt_config), I tried this-
```hcl
read_terragrunt_config(find_in_parent_folders("env.hcl", "i_dont_exist.hcl"), {inputs = {}})
```
This works!!

But it's still a little odd that I have to pass a dummy filename `i_dont_exist.hcl` just to get the fallback params, wouldnt it be better to have something like this-

```hcl
read_terragrunt_config(find_in_parent_folders("env.hcl"), {inputs = {}})
```

**Comment by yorinasub17** [comment]
Ah this is a regression from the conversion to make `find_in_parent_folders` return an absolute path. You need to make the second arg to `find_in_parent_folders` also return an absolute path for this to work. E.g.,  `find_in_parent_folders("env.hcl", "${get_terragrunt_dir()/../../../i_dont_exist.hcl")`.

It probably would help to have a utility function `get_repo_root()` that returns the root of the git repo (or error if not in git repo) to make it easier to construct this path.

Alternatively, does

```hcl
try(
  read_terragrunt_config(find_in_parent_folders("env.hcl")),
  {inputs = {}},
)
```

work? If so, that might also be a cleaner way to handle this.

**Comment by kruttik-lab49** [comment]
Yeah. I already updated my original post with the solution that works. It's very close to what you suggested except that we need to pass a second `"i_dont_exist.hcl"` argument to `find_in_parent_folders` function in order for the `inputs` to kick in.

**Comment by yorinasub17** [comment]
> It's very close to what you suggested except that we need to pass a second "i_dont_exist.hcl" argument to find_in_parent_folders function in order for the inputs to kick in.

It's actually a bit more different than that. Note that I am wrapping the call in the [try](https://www.terraform.io/docs/configuration/functions/try.html) function to handle the error returned by `find_in_parent_folders`. The second arg is actually being passed to `try` so that it can return it when the `read_terragrunt_config` or `find_in_parent_folders` functions fail.

**Comment by mmmorris1975** [comment]
I've run in to a similar issue with find_in_parent_folders().  What's the blocker preventing that from behaving in a way that the preferred and fallback value (if provided) are looked up the same way?  Having to explicitly pass the full path for the fallback seems to go against the spirit of this built-in.  At the very least the docs should make mention of this caveat to warn folks away from thinking the fallback property will be looked up using the same path traversal logic as the primary file.

**Comment by yorinasub17** [comment]
> What's the blocker preventing that from behaving in a way that the preferred and fallback value (if provided) are looked up the same way? 

The fallback is meant to be returned verbatim to handle cases where you don't want to read that value in when it doesn't exist. E.g., you can have it return `""` and check for that in a conditional. Or alternatively, you want to reference a file in the same directory (`find_in_parent_folders` does not check the current directory).

> At the very least the docs should make mention of this caveat to warn folks away from thinking the fallback property will be looked up using the same path traversal logic as the primary file.

This behavior is mentioned in [the function docs](https://terragrunt.gruntwork.io/docs/reference/built-in-functions/#find_in_parent_folders): 

> You can also pass an optional second fallback parameter which causes the function to **return the fallback value** (instead of exiting with an error) if the file in the name parameter cannot be found:

But perhaps it is easy to miss as it is buried in a sentence at the end? Do you have suggestions on how we could improve that verbiage (or placement) so the behavior is more clear?

---

FWIW, I think we could add an enhancement to the function where you can pass in infinite args to the function, and will continue to lookup each arg as a fallback until the last one arg which is returned verbatim.
